packages/common/src/TransactionUtils.ts

```diff
@@ -242,15 +242,15 @@ export async function findBlockNumberAtTimestamp(
   if (higherLimitStamp) {
     if (block.timestamp >= higherLimitStamp) {
       while (block.timestamp >= higherLimitStamp) {
-        blockNumber -= blockDelta;
+        blockNumber -= 1;
```

Comment:
These last two binary searches should always use 1 as the block delta since we are now trying to pinpoint the exact block.

Reply:
I think I agree with this.